generators/unifiedsearch-core/src/test/java/uk/gov/justice/services/clients/unifiedsearch/core/UnifiedSearchTransformerCacheUtilsTest.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.services.clients.unifiedsearch.core;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+
+import uk.gov.justice.services.clients.unifiedsearch.core.domain.UnifiedSearchDescriptor;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+
+public class UnifiedSearchTransformerCacheUtilsTest {
+
+    @Test
+    public void shouldThrowExceptionWhenTransformerPathsAreMoreThanOne() throws MalformedURLException {
+        final List<URL> urls = Arrays.asList(new URL("http://test.com/"), new URL("http://test.com/"));
+        final UnifiedSearchTransformerCacheUtils unifiedSearchTransformerCacheUtils = new UnifiedSearchTransformerCacheUtils();
+        try {
+            unifiedSearchTransformerCacheUtils.verifyTransformerPaths(urls);
+            fail();
+        } catch (final UnifiedSearchException expected) {
+            assertThat(expected.getMessage(), is("You must declare single transformer"));
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenTransformerPathsAreEmpty() throws MalformedURLException {
+        final List<URL> urls = new ArrayList<>();
+        final UnifiedSearchTransformerCacheUtils unifiedSearchTransformerCacheUtils = new UnifiedSearchTransformerCacheUtils();
+        try {
+            unifiedSearchTransformerCacheUtils.verifyTransformerPaths(urls);
+            fail();
+        } catch (final UnifiedSearchException expected) {
+            assertThat(expected.getMessage(), is("You must declare single transformer"));
+        }
+    }
+
+
+    @Test
+    public void shouldThrowExceptionWhenDescriptorsAreMoreThanOne() throws MalformedURLException {
+        final List<UnifiedSearchDescriptor> unifiedSearchDescriptorList = Arrays.asList(mock(UnifiedSearchDescriptor.class), mock(UnifiedSearchDescriptor.class));
+        final UnifiedSearchTransformerCacheUtils unifiedSearchTransformerCacheUtils = new UnifiedSearchTransformerCacheUtils();
+        try {
+            unifiedSearchTransformerCacheUtils.verifyUnifiedSearchDescriptor(unifiedSearchDescriptorList);
+            fail();
+        } catch (final UnifiedSearchException expected) {
+            assertThat(expected.getMessage(), is("You must declare single unified-search-descriptor.yaml"));
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenDescriptorsAreEmpty() throws MalformedURLException {
+        final List<UnifiedSearchDescriptor> unifiedSearchDescriptorList = new ArrayList<>();
+        final UnifiedSearchTransformerCacheUtils unifiedSearchTransformerCacheUtils = new UnifiedSearchTransformerCacheUtils();
+        try {
+            unifiedSearchTransformerCacheUtils.verifyUnifiedSearchDescriptor(unifiedSearchDescriptorList);
+            fail();
```

Comment:
You may want to look at the `ExpectedException` class?

Reply:
done